### ISSUE TITLE
Semantic-UI web components

### ISSUE THREAD

**Issue by caitp** [Semantic-Org/Semantic-UI#565]
So, cool things are happening with Web Components these days, FF landed a lot of support (pref'd off by default) in December, Chromium has had some experimental web component support for ages now (pref'd off by default), and Safari is in a similar boat. It's getting close to being usable completely by default on the web.

That's all great and all, but people want to use this stuff right now! I think working with Polymer, it should be fairly trivial to turn all of the Semantic-UI components into proper web components, and should allow the project to be a lot lighter, too.

Cool things that this would afford:
- among others, removing any dependency on jQuery and making use of the framework much more declarative, which is really cool.
- much simpler interoperability with outside frameworks (like Ember, AngularJS, etc)

I realized, since I'm not putting too much work into angular-semantic anymore, that really all it's doing is trying to make some nice declarative components which play nicely with the Angular ecosystem. That's a cool thing and all, but it could be a whole lot cooler if it was designed around components to begin with.

That's all a pretty tall order, but lets face it, there are issues with the code as it stands:
- Creates absolutely horrific markup, div soup
- Requires some imperative logic to make anything actually "work"

I think this could be done a lot better. Hide the div-soup behind the shadow-DOM, let users write proper semantic markup that is easily understood

This sounds like a complaint, but it's not. This sounds like a feature request, and I guess in a way it is. It means re-writing and re-organizing a lot of code, and I think that's okay. It means simplifying and improving the performance of a lot of code, and I think that's always a good thing.

So, I think some thought should be given to turning semantic-ui into something which plays nicely with the web components specifications, and which gives programmers and designers the chance to work with truly semantic markup. Keep the issue open as a reminder of what the web is starting to look like now that these features are widespread.

**Comment by anunos** [comment]
Hello there, @jlukic @caitp any intentions to reopen this issue now that Polymer 1.0 is out and production ready?

**Comment by caitp** [comment]
it would be cool, and Blink is improving the CSS encapsulation of their components fairly frequently (I believe shadow piercing selectors landed a few days ago, or will be landing soon). So good things are happening there --- But, it might be a bit crazy to refactor the entire framework just for those niche benefits. Maybe a "semantic-component" repo could be opened for some experimental work in that arena.

**Comment by jlukic** [comment]
I'm pretty much of the same mindset as @caitp. 

I'm not confident yet that the front end development community will embrace the shadow/logical DOM and its idiosyncratic selector engine. Swapping everything over to web components may mean I risk making a framework that only a few people would know how to modify or contribute to.

**Comment by kapouer** [comment]
I started using Custom Elements and Link imports on a new website with a cms. It is very useful for development and clean separation of components, and it is very easy to setup.
CSS componentization wasn't possible in that case because we started from a static website, written beforehand using another framework (bootstrap). The custom elements we defined are things like sliders, figures with caption, whole menus, not "small" elements like the Material custom elements - working with those feels like learning a whole new set of constraints.
To sum up, i don't think semantic-ui would benefit from those (maybe in a few cases like the progress module) but it certainly shouldn't get in the way of using them !

**Comment by ansarizafar** [comment]
1+ for Semantic ui for Web components (Polymer). Is there any update?
